### PR TITLE
Adds per-client consent customization

### PR DESCRIPTION
**Pull request by ka7eh**
Two new boolean fields are added to the client model: `skip_consent` and `require_consent`. They replace `OIDC_SKIP_CONSENT_ENABLE` and `OIDC_SKIP_CONSENT_ALWAYS` respectively.

The new fields allow customization of skipping or requiring consent per client.

### CODE REVIEW

**Comment by wojtek-fliposports** [comment]
I will review this today. 
Thank You for collaborating. I think this solves #130

**Comment by ka7eh** [comment]
No problem, will take care of it.

**Comment by ka7eh** [comment]
I updated two sections in the docs: `settings` and `relyingparties`. Is there any other place that might be referencing the removed settings?

**Comment by wojtek-fliposports** [comment]
I will check that tomorrow but meanwhile can You add also information about removed settings into changelog file ?

**Comment by ka7eh** [comment]
Sure, just updated the PR.

File: oidc_provider/models.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ class Client(models.Model):
     terms_url = models.CharField(max_length=255, blank=True, default='', verbose_name=_(u'Terms URL'), help_text=_(u'External reference to the privacy policy of the client.'))
     contact_email = models.CharField(max_length=255, blank=True, default='', verbose_name=_(u'Contact Email'))
     logo = models.FileField(blank=True, default='', upload_to='oidc_provider/clients', verbose_name=_(u'Logo Image'))
+    skip_consent = models.BooleanField(default=True, verbose_name=_('Skip Consent?'), help_text=_('If enabled, the Server will save the user consent given to a specific client, so that user won\'t be prompted for the same authorization multiple times.'))
```
Comment (clintonb) [inline oidc_provider/models.py]: This should default to `False` to avoid accidentally creating new clients with consent skipped. Better safe than sorry.

Comment (clintonb) [inline oidc_provider/models.py]: Can this be renamed to something less-confusing? "Skip" implies (to me, at least) that consent is always skipped. Based on the help text this is more of a `one_time_consent` or some other name indicating that consent is received only once.

File: setup.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
     test_suite='runtests.runtests',
     tests_require=[
         'pyjwkest==1.3.0',
-        'mock==2.0.0',
```
Comment (clintonb) [inline setup.py]: Nit: Restore the trailing comma.

File: oidc_provider/models.py (hunk 1 above)
Comment (wojtek-fliposports) [inline oidc_provider/models.py]: i was "skip consent" before and basically it's problem of "naming things". I can accept this PR but I think that @juanifioren should make final decision.

Comment (ka7eh) [inline oidc_provider/models.py]: What about `reuse_consent`? I think it implies the purpose better than other names.

Comment (wojtek-fliposports) [inline oidc_provider/models.py]: Can you change this to `reuse_consent` ? and push changes to this branch ?

PS. Great contribution. Thanks.

Comment (ka7eh) [inline oidc_provider/models.py]: The field name is updated. Let me know if there's anymore changes needed. Cheers

**Comment by ka7eh** [comment]
...and squashed.

**Comment by ka7eh** [comment]
I just noticed this PR was against the `v0.4.x` branch. Should it be merged with the `develop` branch too?

**Comment by wojtek-fliposports** [comment]
yes.. 
you got me :D

**Pull request merged**